Docs/source/glossary.rst

```diff
@@ -0,0 +1,232 @@
+
+  Op-Amp
+    An operational amplifier is a circuit element that performs mathematical operations on input voltages. It is most commonly used to increase these inputs by a certain magnitude to make them easier to read by the microcontroller.
+
+  Phase Locked Loop
+    The PLL generates an output signal whose phase is related to the input signal. It is used to increase the clock speed of the STM microcontroller.
+
+  Positive Logic 
+    Logical 1/true is a high voltage/on and a logic 0/false is a low voltage/false. 
+
+  Printed Circuit Board
+    PCB's are copper circuits printed onto non-conductive boards that electrically connect our electronic
+    components together and allow us hardware to run our programs on.
+
+  Pull-Down Resistor 
+    A resistor to gnd that pulls a wire's voltage to logical '0' when it disconnects from some other source. 
+
+  Pull-Up Resistor 
+    A resistor to Vcc that pulls a wire's voltage to logical '1' when it disconnects from some other source. 
+
+  Pulse Width Modulation
+    A method of controlling the power usage of a peripheral by quickly turning it on and off and controlling how long it stays on for. Used to control the speed of the fans.
+
+  Push-Pull 
+    A configuration of a pin on a microcontroller where typically logical '0' corresponds to 0V, which logical '1' corresponds to the operating voltage (Vcc). Using this configuration, there can only be one source on a wire that drives communication/control. It stands in opposition to an "open drain" configuration. 
+
+  Queue 
+    A First-In-First-Out (FIFO) structure, where typically supported operations are to insert elements (queue/push), take elements out (dequeue/pop), and view the front element (peek). Elements are removed in the same order that they were entered. 
+
+  Race Condition 
+    An issue that occurs when multiple threads are attempting to access the same resources. For example, if one thread performs an operation on some data that takes the form read-modify-writeback, and another thread writes to that data after the first read, but before it modifies and writes back. This could potentially put the program in an unaccounted-for state. This issue can be addressed with mutexes and semaphores. 
+
+  Register 
+    Basic unit of memory. Temporarily holds important data. 
+
+  RTOS 
+    A real-time operating system. These typically manage threads in a way that requires each thread to cede control back to the operating system, rather than allowing the OS to interrupt the thread. This ensures that communications and other critical paths are not interrupted partway through completion. These may or may not include other features like file systems, networking stacks, and memory management systems. 
+
```

Comment:
Should we include that the main BPS uses an RTOS while the supplemental board is just a bare metal loop

Reply:
This is documentation for just the Sunlight subsystem. I copied over the glossary from the BPS documentation because it is much of the same stuff. The supplemental board documentation will probably be added to the BPS documentation.